### PR TITLE
Add FASTQ_ALIGN_DNA subworkflow

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]
I think for snapaligner we might need to do some more checks and docs first? My understanding is we should then neither do splitting nor run duplicate marking.

**Comment by maxulysse** [comment]
For the moment, neither snapaligner nor bowtie2 can be used because I haven't add them in our params.aligner

**Comment by matthdsm** [comment]
cfr snapaligner
It's able to handle both, but you can disable it with settings. Splitting fastq's is not an issue, but that's of course a bad idea if you want to use the included duplicate marking.

our current config looks like this
```
    // -b-    : Don't lock threads to cpu cores, improves performance when scheduling multiple jobs/node
    // -sm 10 : Allow 10GB of memory for sorting threads

    withName: SNAP_ALIGN {
        cpus   = { check_max( meta.tag && meta.tag == "WGS" ? 36 : 18 , "cpus" ) }
        memory = { check_max( meta.tag && meta.tag == "WGS" ? 252.GB : 126.GB , "memory" ) }
        ext.when = {params.aligner == "snapaligner"}
        ext.args = {[
            "-so",
            "-b-",
            "-sm 10",
            meta.readgroup ? "-R \"@RG\\t" + meta.readgroup.collect{ "$it.key:$it.value" }.join("\\t") + "\"" : ""
        ].join(" ").trim()}
        publishDir = [ enabled: false ]
    }
```

**Comment by FriederikeHanssen** [comment]
> For the moment, neither snapaligner nor bowtie2 can be used because I haven't add them in our params.aligner

Probably late to the party, but have to say not a big fan of having dead code in the pipeline, it makes the whole thing less readable imo. Maybe let's discuss this later today

**Comment by maxulysse** [comment]
> > For the moment, neither snapaligner nor bowtie2 can be used because I haven't add them in our params.aligner
> 
> Probably late to the party, but have to say not a big fan of having dead code in the pipeline, it makes the whole thing less readable imo. Maybe let's discuss this later today

Not dead code, but a placeholder for future features

**Comment by matthdsm** [comment]
Also, there's no index for snap in iGenomes!
hint hint @FriederikeHanssen 😉 
https://nfcore.slack.com/archives/CELEMAGC9/p1657108775156149

File: workflows/sarek.nf [hunk 1]
```diff
@@ -344,8 +344,8 @@ workflow SAREK {
     rt_file                = PREPARE_GENOME.out.rt_file
 
     // Gather index for mapping given the chosen aligner
-    ch_map_index = params.aligner == "bwa-mem" ? bwa :
-        params.aligner == "bwa-mem2" ? bwamem2 :
+    ch_map_index = params.aligner == "bwamem" ? bwa :
+        params.aligner == "bwamem2" ? bwamem2 :
```
Comment (matthdsm) [inline workflows/sarek.nf:348]: might have to add `bowtie2` and `snap` here too

**Comment by matthdsm** [comment]
I can't see any aligner tests in the CI. Am I right?

File: workflows/sarek.nf (hunk 1 above)
Comment (maxulysse) [reply workflows/sarek.nf:348]: I'll do it in another PR, need to add the index as well, and docs, and probably test this.

**Comment by maxulysse** [comment]
> I can't see any aligner tests in the CI. Am I right?

https://github.com/nf-core/sarek/actions/runs/3240467988/jobs/5311130371

**Pull request closed without merging**